src/utils/hooks/useGetApplicationStatus.tsx

```diff
@@ -1,65 +0,0 @@
-import { useEffect, useState } from 'react'
-import { ApplicationStages, StageAndStatus, UseGetApplicationProps } from '../types'
-import {
-  Application,
-  ApplicationStageStatusAll,
-  TemplateStage,
-  useGetApplicationStatusQuery,
-} from '../generated/graphql'
-import useTriggerProcessing from './useTriggerProcessing'
-
-const useGetApplicationStatus = ({
-  serialNumber,
-  isApplicationLoaded,
-  networkFetch,
-}: UseGetApplicationProps) => {
-  const [appStages, setAppStages] = useState<ApplicationStages>()
-  const [appStatus, setAppStatus] = useState<StageAndStatus>()
-
-  const { triggerProcessing, error: triggerError } = useTriggerProcessing({
-    serialNumber,
-    // triggerType: 'applicationTrigger',
-  })
-
-  const { data, loading, error } = useGetApplicationStatusQuery({
-    variables: {
-      serial: serialNumber,
-    },
-    skip: triggerProcessing || !isApplicationLoaded,
-    fetchPolicy: networkFetch ? 'no-cache' : 'cache-first',
```

Comment:
I think this is causing the problem I mentioned above. The status needs to ensure it gets the fresh value from the network immediately after submission, hence the dynamic fetchPolicy here. You'll need to do something similar again to the `useGetApplicationQuery` in `useLoadApplication` now that you've consolidated them into one Hook.

Reply:
Very nice! I wasn't sure what was causing this either. Will fix now 👍